code/modules/vehicle/vehicle.dm

```diff
@@ -238,3 +238,10 @@
 /obj/vehicle/zap_act(power, zap_flags)
 	zap_buckle_check(power)
 	return ..()
+
+/obj/vehicle/water_act(volume, temperature, source, method)
+	. = ..()
+	if(HAS_TRAIT(src, TRAIT_OIL_SLICKED))
+		vehicle_move_delay = initial(vehicle_move_delay)
```

Comment:
I think you might want to increment and decrement vehicles by a fixed amount when adding/removing this trait, this might end up resetting something extraneous.

Reply:
Removed